{"events":[{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","event":"call","id":179853,"lineno":84,"method_id":"handleDatabaseIssue_cannotCreateTx","path":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","object_id":1195771493,"value":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues@47460665"},"static":false,"thread_id":1},{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandler","event":"call","id":179854,"lineno":81,"method_id":"handleDatabaseConnectionIssue","parameters":[{"class":"org.springframework.transaction.CannotCreateTransactionException","kind":"req","name":"ex","object_id":359774579,"value":"org.springframework.transaction.CannotCreateTransactionException: db down"}],"path":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandler","object_id":1032063310,"value":"com.example.accesscontrol.exception.GlobalExceptionHandler@3d84094e"},"static":false,"thread_id":1},{"elapsed":0.0,"event":"return","id":179855,"parent_id":179854,"return_value":{"class":"org.springframework.http.ResponseEntity","object_id":1542915389,"value":"<503 SERVICE_UNAVAILABLE Service Unavailable,{timestamp=2025-08-16T11:22:12.308864300, error=Database connection failed. Please try again later., status=503},[]>"},"thread_id":1},{"elapsed":1.0029E-4,"event":"return","id":179856,"parent_id":179853,"thread_id":1}],"classMap":[{"children":[{"children":[{"children":[{"children":[{"children":[{"labels":[],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:84","name":"handleDatabaseIssue_cannotCreateTx","static":false,"type":"function"}],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","name":"GlobalExceptionHandlerTest$DatabaseIssues","static":false,"type":"class"},{"children":[{"labels":[],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java:81","name":"handleDatabaseConnectionIssue","static":false,"type":"function"}],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","name":"GlobalExceptionHandler","static":false,"type":"class"}],"name":"exception","type":"package"}],"name":"accesscontrol","type":"package"}],"name":"example","type":"package"}],"name":"com","type":"package"}],"version":"1.2","metadata":{"name":"Global exception handler $ database issues handle database issue _ cannot create tx","app":"accesscontrol","language":{"name":"java","version":"17.0.15+6-LTS","engine":"OpenJDK 64-Bit Server VM"},"client":{"name":"appmap-java","url":"https://github.com/appland/appmap-java"},"recorder":{"name":"junit","type":"tests"},"recording":{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","method_id":"handleDatabaseIssue_cannotCreateTx"},"git":{"repository":"https://github.com/AlhaderMajd/accessControl-majd.git","branch":"main","commit":"9e16780f0e1a4f79a3317f0154c9ec87f7bf0759"},"source_location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:84","frameworks":[{"name":"JUnit","version":"5"}],"test_status":"succeeded"},"eventUpdates":{}}